packages/sharing-smart-contract/test/consumeProtectedData.test.js

```diff
@@ -27,34 +30,105 @@ describe('ConsumeProtectedData', () => {
+      addr1,
+      addr2,
+    } = await loadFixture(createAssets);
+    const tx = await protectedDataSharingContract.connect(addr1).createCollection();
+    const receipt = await tx.wait();
+    const collectionTokenId = ethers.toNumber(receipt.logs[0].args[2]);
 
-      const { iexecWorkerpoolOwner, workerpoolAddress } = await createWorkerpool(rpcURL);
+    // add protectedData to Collection
+    const registry = await ethers.getContractAt(
+      'IRegistry',
+      '0x799daa22654128d0c64d5b79eac9283008158730',
+    );
+    const protectedDataTokenId = ethers.getBigInt(protectedDataAddress.toLowerCase()).toString();
+    await registry
+      .connect(addr1)
+      .approve(await protectedDataSharingContract.getAddress(), protectedDataTokenId);
+    await protectedDataSharingContract
+      .connect(addr1)
+      .addProtectedDataToCollection(collectionTokenId, protectedDataAddress, appAddress);
 
-      // create fake workerpoolOrder
-      const workerpoolOrder = await createWorkerpoolOrder(iexecWorkerpoolOwner, workerpoolAddress);
+    // set up subscription
+    const subscriptionParams = {
+      price: ethers.parseEther('0.5'),
+      duration: 2_592_000, // 30 days
+    };
+    await protectedDataSharingContract
+      .connect(addr1)
+      .setSubscriptionParams(collectionTokenId, subscriptionParams);
+    await protectedDataSharingContract
+      .connect(addr1)
+      .setProtectedDataToSubscription(collectionTokenId, protectedDataAddress);
 
-      await setAppAddress(protectedDataSharingContract, appAddress);
+    // set up renting
+    const rentingParams = {
+      price: ethers.parseEther('0.7'),
+      duration: 172_800, // 2 days
+    };
+    await protectedDataSharingContract.setProtectedDataToRenting(
+      collectionTokenId,
+      protectedDataAddress,
+      rentingParams.price,
+      rentingParams.duration,
+    );
+    return {
+      protectedDataSharingContract,
+      protectedDataAddress,
+      workerpoolOrder,
+      collectionTokenId,
+      subscriptionParams,
+      rentingParams,
+      addr2,
+    };
+  }
 
-      const { tx } = await consumeProtectedData(
+  describe('ConsumeProtectedData()', () => {
+    it('should create a deal on chain if an end user subscribe to the collection', async () => {
+      const {
         protectedDataSharingContract,
-        contentAddress,
+        protectedDataAddress,
         workerpoolOrder,
-        '',
-      );
+        collectionTokenId,
+        subscriptionParams,
+        addr2,
+      } = await loadFixture(createCollection);
+
+      await protectedDataSharingContract.connect(addr2).subscribeTo(collectionTokenId, {
+        value: subscriptionParams.price,
+      });
+
+      const tx = await protectedDataSharingContract
+        .connect(addr2)
+        .consumeProtectedData(collectionTokenId, protectedDataAddress, workerpoolOrder, '');
+      await tx.wait();
 
       expect(tx)
         .to.emit(protectedDataSharingContract, 'DealId')
```

Comment:
`DealId` is a strange name for an event
We may want to rename it in `ProtectedDataConsumed` and add `mode: subscription | renting`  for the sake of consumption indexing

Reply:
I do it : Commit: 117816743a7bd9ab657800bdf41a050482a8565b
But i didn't find that very useful, and it consume more gas